android/library/core/src/main/java/com/liferay/mobile/screens/base/BaseScreenlet.java

```diff
@@ -92,14 +93,28 @@ public I getInteractor(String actionName) {
 		return result;
 	}
 
+	public void addCustomInteractor(CustomInteractorListener customInteractorListener) {
+		_customInteractorListener = customInteractorListener;
+	}
+
 	protected I prepareInteractor(String actionName) {
+
+		if (_customInteractorListener != null) {
+			I customResult = (I) _customInteractorListener.createInteractor(actionName);
+
+			if (customResult != null) {
+				customResult.onScreenletAttached(this);
```

Comment:
Code repeated. Get interactor from custom listener (if exists), else get it from template method. For both, attach, put and return